addons/addon-base-raas/packages/base-raas-services/lib/data-source/data-source-account-service.js

```diff
@@ -238,13 +238,39 @@ class DataSourceAccountService extends Service {
         return op.item(item).update();
       },
       async () => {
-        throw this.boom.notFound(`The data source account entity "${arn}" does not exist`, true);
+        throw this.boom.notFound(`The data source account entity "${id}" does not exist`, true);
       },
     );
 
     return toDsAccountEntity(dbEntity);
   }
 
+  async updateStackCreated(requestContext, { stackCreated, dataSourceAccount } = {}) {
```

Comment:
To be consistent with the rest of the file,  can you rename dataSourceAccount to dsAccountEntity?

Reply:
Or you can just pass { id, stackCreated } = {}
Then in the accountIdCompositeKey.encode(dataSourceAccount) you can do this instead
accountIdCompositeKey.encode({ id } )

Reply:
Done